### ISSUE TITLE
Plugin Settings Page memory leak in installation with lots of assets

### ISSUE THREAD

**Issue by mnlmaier** [heavymetalavo/craft-aialttext#15]
Hi,

thanks for the work on this plugin. We're currently evaluating its usage in a project with about 100k+ assets. Sadly, I am unable to configure the plugin, as the AI Alt Text settings page in the control panel (`/admin/settings/plugins/ai-alt-text`) will not load. When trying to access the site, the only thing that happens is an error message that the `memory_limit` of my PHP installation has been exceeded, no matter what I set it to. When setting it to `-1`, ie unlimited, I can notice the following:

![Image](https://github.com/user-attachments/assets/dae83d03-0b5c-4638-b346-c4cd1d6239ed)

Memory usage is going up over time and will only be capped by the capabilities of my laptop — seems like a memory leak.

As I would really like to continue evaluating the plugin, do you have any idea what might cause this?

---

Edit: Also happening in PHP 8.3, though the memory usage creeps up a little slower.

**Comment by joepagan** [comment]
Hey @mnlmaier thanks for trying the plugin and reporting your issue. That's a lot of assets! I have not yet built in contingencies for sites with high numbers like that.
It's probably the queries which build the table data on that settings page. 
I'll take a look for you and see if I can get that working.

**Comment by mnlmaier** [comment]
@joepagan — awesome, thanks for the quick response! yep, I know that this is not a usual amount of assets, but for obvious reasons, automated AI alt texts would be really helpful in this project! 😄  thanks a lot!

**Comment by joepagan** [comment]
Hi @mnlmaier just letting you know I've released [v1.5.8](https://github.com/heavymetalavo/craft-aialttext/releases/tag/v1.5.8) which should work for you now.
I tested a ~50k asset project with a `64M` php memory_limit and could load the settings page no problem. There _may_ be further optimisations on the bulk actions logic which generate the queue jobs. It managed to get to generate about 12k before nginx timing out for me on a pretty standard ddev setup.
Can you let me know if that works for you now? 
Appreciate this isn't the best for generating all the jobs at once, but you should be able to get through a decent chunk over several requests potentially. You may have to just run that a few times.

I'm working on a new feature for adding console commands for the bulk actions which may benefit you! Hopefully I will be able to release that within this week.

**Comment by joepagan** [comment]
Hi @mnlmaier Just letting you know [v1.6.0](https://github.com/heavymetalavo/craft-aialttext/releases/tag/v1.6.0) is now released which includes console commands - Hopefully that helps you process all your assets easier!
If you get an opportunity to test your original issue again I'd love to know if it is now resolved so we can close this issue.

**Comment by mnlmaier** [comment]
@joepagan — sorry, I got busy with other tasks, but will get back to evaluating the plugin now. thank you for your effort and the very quick reply, this is very much appreciated! I'll let you know how it goes and close the issue, if everything is working fine now :)

**Comment by mnlmaier** [comment]
Following up on this: Loading times have improved and the plugin is now usable :)
